### DISCUSSION TITLE
Is it possible to create pre-set markers onclick, rather than drag & selecting an area?

### DISCUSSION THREAD

**Discussion by gbrits** [annotorious/annotorious#259] [Q&A]
As per the title of the question, I have a client that has asked me if it's possible to just click on the image and it creates a preset, let's say 100x100 shape on the click, instead of requiring a select (click, drag & end) as is the default function?

Thanks.

**Comment by rsimon** [comment]
Hi, not out of the box. I'd need to play with this myself to be sure... But I think the least painful hack would be to set Annotorious to `readOnly` mode, and then do your own mouse click tracking on the image or the annotation layer element. If the user clicks, you could generate your own annotation object, e.g. something like this:

```js
{ 
  "@context": "http://www.w3.org/ns/anno.jsonld",
  "id": "#my-random-id",
  "type": "Annotation",
  "body": [],
  "target": {
    "selector": {
      "type": "FragmentSelector",
      "conformsTo": "http://www.w3.org/TR/media-frags/",
      "value": "xywh=pixel:{mouse-x},[mouse-y},{fixed-width},{fixed-height}"
    }
  }
}
```

and then add this annotation via `anno.addAnnotation`. The only problem is that they would stay read-only. So you'd need to have some kind of toggle logic. Could get hacky...

The other alternative - althought that would be more involved, would be to build your own drawing tool. You could start by looking into the implementation of the [normal Rubberband Rectangle tool here](https://github.com/recogito/annotorious/blob/main/src/tools/rectangle/RubberbandRectTool.js), and perhaps build a simpler version of it, which just places a fixed-size rectangle.
